### DISCUSSION TITLE
Kaplan-Meier(KM) plot on safety with (any) categorical factor

### DISCUSSION THREAD

**Discussion by mli1** [SafetyGraphics/safetyGraphics#496] [Chart Discussion]
# Name of the chart
{ KM plot on safety with any categorical factor }

# Objectives 
{ To visualize the consistency (or not) among subgroups of a giving categorical factor with regard to: 1)  the time of onset of a particular AE of interest (or AEs) ;  2)- the proportion of subjects who developed the AE (s)   }

# User Interface mockup
{
![KM plot example on safety with categorical stratification factor](https://user-images.githubusercontent.com/41388538/107535849-22023f00-6b8f-11eb-9c18-d75e8e308bec.PNG)
 } 

# Features of the chart 
{ We can start with static, but it could be easily made in an interactive one by choosing different categorical factor; might also make the AE of interest to be another variable that we can choose from }

# Data Requirement;
{ Will add later }

# References
{ Provide links to related papers or examples } 

Idea from Jerry and Mengchun @tbajn @mli1

**Comment by JimBuchanan** [comment]
Personally, I find the KM plot the best way to visualize a risk period for development of an AE, particularly when compared to another treatment or control group. You can accomplish the same thing by plotting the risk ratio over time, but I find the KM plot easier to see. Interactivity can be made in several ways. One is to be able to scroll through a series of AEs to find the ones with an apparent separation of the KM curves. Or there could be some programming to identify just those AEs demonstrating a curve separation of > "X". Having then identified the AEs showing a curve separation, you could then ask the question whether the effect is being driven by any underlying characteristic. Thus, plot out the curves by subgroups (e.g., gender, age range, presence/absence of underlying medical conditions, presence/absence of a concomitant medication, etc.). And like in the volcano plot, you should be able to list out the patients with that AE of interest so you could dig further.

**Comment by xni7** [comment]
Great ideas @mli1 @tbajn and @JimBuchanan ! A few cents:

- Jim (and @mli1 in another thread) raised a key idea on the workflow.  Like the interactive eDISH, it would be useful to start with an overview page to help users identify the signal and further drill down. It can be scanning through all AEs, or identify subgroups for prespecified AE of special interest. 
- Some existing options of summary graph (**interactive** landing page, to drill down from)  include
   + Tendril plot, this includes time to **recurrent** events, see [Tendril R package vignette](https://cran.r-project.org/web/packages/Tendril/vignettes/TendrilUsage.html)
![image](https://user-images.githubusercontent.com/31833867/107878556-72381480-6ea1-11eb-80de-68372d4fdf5a.png)
   +  AE dot plot 
![image](https://user-images.githubusercontent.com/31833867/107878425-93e4cc00-6ea0-11eb-9ff4-0b7389abdf78.png)
   + Volcano plot ![image](https://user-images.githubusercontent.com/31833867/107878515-3735e100-6ea1-11eb-9ff3-c1369f2a1647.png)
- For the original objective of detecting subgroups statistically, one idea might be to repurpose the Volcano plot, showing the **hazard ratio** on one axis and (adjusted) p value of log-rank test on the other. **The limitation** may be for lower frequency and smaller sample sizes it would be difficult to draw inference. Certainly we could add some dropdowns to allow users inspect subgroups one or several at a time. 
- Finally for the original objective of showing incidence, there are existing tools/graphs in R. One could reuse these pkgs and get the numerical summarize below the KM curves, or one can make the KM curve interactive so that users can hover over a point on the curve to see the numbers.
    + [See this site](https://www.emilyzabor.com/tutorials/survival_analysis_in_r_tutorial.html#Add_the_numbers_at_risk_table)
![image](https://user-images.githubusercontent.com/31833867/107878820-140c3100-6ea3-11eb-9571-4c6790c5d85b.png)
    + and [this tutorial](http://www.sthda.com/english/wiki/survminer-r-package-survival-data-analysis-and-visualization#transform-survival-curves-plot-cumulative-events-and-hazard-function) has similar plots as above, but offers an idea to **overlay subgroups on one plot**
![image](https://user-images.githubusercontent.com/31833867/107878906-b2989200-6ea3-11eb-8677-9827181390a7.png)

**Comment by JimBuchanan** [comment]
Good thoughts. And expanding on your suggestion to repurpose the Volcano plot to visualize subgroups, here is an example of that based on subgrouping by gender:
![image](https://user-images.githubusercontent.com/38111134/107980168-a4ff0d00-6f74-11eb-84c5-33a685303707.png)
I also like the Tendril plot option. When you say it can display recurrent events, though, are you referring to recurrence within a single patient as I'm not sure how that would appear in the plot since it would seem to be displaying the group aggregate data.

**Comment by tbajn** [comment]
Hi all -- just to add that an additional covariate whose influence on time to first AE occurrence can be explored in this way is exposure. One categorizes some exposure metric, such as steady-state AUC, into quantiles, and then plots KM curves by quantile. Below is an example. In my experience, the AEs examined in this way are typically either prespecified because of known safety signals for the treatment under investigation, or it is prespecified that, say, the top 5 most frequent AEs will be examined in this way. 
![image](https://user-images.githubusercontent.com/77704625/108366087-f4884780-71c5-11eb-9bf8-951d1cf7a8f0.png)

**Reply by JimBuchanan** [reply]
That's an interesting use of KM curves. I like it. I can see in SAD and MAD studies where you are collecting PK you could use AUC or Cmax. Then in Phase 2 studies using more than one dose you could parse it by dose level. Nice!

**Comment by tbajn** [comment]
Thanks. The example shown was actually from Phase 3 where PK samples were collected and the AUCs were derived from a population pharmacokinetic model, so that's another possibility.
